### DISCUSSION TITLE
Cable of cables? How to group cables?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [netbox-community/netbox#5749] [Q&A] [answered]
I've setup a few patch panels and connected the rear ports of the, connected devices, etc. Works fine.

I'm wondering though, how do you group the cables connected to the rear ports into 24/48 strand fiber? Is there some sort of grouping of cables? I know at the most granular level, every strand is a cable as far as netbox is concerned, I'd just like to be able to group them in a way that represents their physical "grouping" as well.

Thanks.

(and sorry if there's a duplicate post)

**Comment by candlerb** [comment, marked as answer]
> I've setup a few patch panels and connected the rear ports of the, connected devices, etc. Works fine.
> 
> I'm wondering though, how do you group the cables connected to the rear ports into 24/48 strand fiber? Is there some sort of grouping of cables?

When you create patch panel, a single "rearport" can be linked to multiple "frontports".  In that case, the cable going into the rearport implicitly carries multiple signals for all of its linked frontports.  You must connect it to another rearport with an equal number of positions.  This makes it very easy to link (say) two 24-port ODFs together, with one cable between them.

See [this post](https://groups.google.com/g/netbox-discuss/c/8EtW5u4fpOM/m/8qWoWWBKBQAJ) and surrounding thread for an example.

Although a rearport-to-rearport connection is just a single "cable", the path tracing logic is smart enough to keep track of the offset within that cable, so it pops out of the correct frontport at the other end.

> I know at the most granular level, every strand is a cable as far as netbox is concerned

That's not actually true.  Netbox does not model strands at all.  A "Cable" in Netbox is an entity with two ends, where each end occupies an Interface or Port or Circuit Termination, and its internals are not modelled beyond type, color and length.

The most common issue this causes is when you're using bog-standard SFPs with separate receive and transmit connections.  Netbox models the interface as "SFP", but doesn't model the SFP itself.  The cable goes directly into the interface, and the interface accepts only one cable - even though this is actually two strands.  As long as your whole infrastructure is using duplex optics then you can model every fibre *pair* as a "cable" and not worry about it, but it breaks down when you mix in BiDi or GPON optics which use single strands.

Netbox also doesn't distinguish between (physical) ports and (electronic) interfaces.  In a more extreme case, you can have a single QSFP port which carries 4 interfaces, and when you put in QSFP module then it may connect to 8 fibre strands, and you may insert a breakout cable with one connector at one end and 4 connectors at the other.  Netbox doesn't cope with any of this well, but this is [by design](https://netbox.readthedocs.io/en/stable/#keep-it-simple).

**Reply by DanSheps** [reply]
@lonelyadmin If @candlerb's answer makes sense, can you please mark it as answered?

**Reply by subbergunz** [reply]
> it breaks down when you mix in BiDi or GPON optics which use single strands

I am facing this exact situation, many dark fiber circuits and a mix of LX/ZX/LR/ER (two strands) and BX (one strand). I have no better clue than making some circuits with one fiber (eg 1, 2, 3) and some circuits with two fibers (eg 5-6, 11-12). When changing type of connection from ZX to BX, delete on circuit and create two (connecting only one of them), and when changing from BX to ZX delete the two circuits and create one.

This is quite ugly. If someone has a better workaround or idea I would be happy to know.

Regards,
                       Bergonz
